viewing_party/party.py

```diff
@@ -1,23 +1,126 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie_dictionary = {}
+    if title and genre and rating:
+        movie_dictionary["title"] = title
+        movie_dictionary["genre"] = genre
+        movie_dictionary["rating"] = rating
+    else:
+        return None
+    
+    return movie_dictionary
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+
+    return user_data
+
+def watch_movie(user_data, title):
+    copy_user_data = user_data
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            copy_user_data["watched"].append(movie)
+            copy_user_data["watchlist"].remove(movie)
+        user_data = copy_user_data
+
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    avg_rating = 0.0
+    sum = 0
+    length = len(user_data["watched"])
+
+    if user_data["watched"]:
+        for movie in user_data["watched"]:
+            sum += movie["rating"]
+        avg_rating = sum / length
+    
+    return avg_rating
+def get_most_watched_genre(user_data):
+    dict_count_genre = {}
+    if user_data["watched"]:
+       for movie in user_data["watched"]:
+           if not movie["genre"] in dict_count_genre:
+              dict_count_genre[movie["genre"]] = 1
+           else:
+              dict_count_genre[movie["genre"]] += 1
+       maximum = max(dict_count_genre, key= dict_count_genre.get)
+       return maximum
+    return None
+
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
-# -----------------------------------------
+def get_unique_watched(user_data):
+    unique_watched = []
+    for movie in user_data["watched"]:
+        title = movie["title"]
+        for friend in user_data["friends"]:
+            if title in list(movie["title"] for movie in friend["watched"]):

```

Comment:
This syntax here `list(movie["title"] for movie in friend["watched"])` looks like we're using a generator and then casting the return value from it to a list. We could use list comprehension syntax here like: `[movie["title"] for movie in friend["watched"]]` which is more straight forward.

What might be more clear is to create a helper function that returns a list of all the movies a user's friends have watched and then call that helper here so that you have something like `friends_movies_list` to iterate over. 

However, if you have `user_data["watched"]` and a `friends_movies_list`, could you convert those lists to sets and find the difference in order to return `unique_watched` ?